src/app/course/leader-board/leader-board.component.ts

```diff
@@ -1,40 +1,78 @@
+// Angular Imports
 import {AfterViewInit, Component, Input, OnInit, ViewChild} from '@angular/core';
 import {MatTableDataSource} from '@angular/material/table';
 import {MatSort} from '@angular/material/sort';
 
+// Model Imports
+
 @Component({
     selector: 'app-leader-board',
     templateUrl: './leader-board.component.html',
     styleUrls: ['./leader-board.component.scss']
 })
+
 export class LeaderBoardComponent implements OnInit, AfterViewInit {
 
-    @Input() leaderBoard: [{
+    // Array of objects
+    // Storing the top three users and their number of tokens
+    topThree: {
+        name: string,
+        token: number
+    } [] = [];
+
+    // Array of objects
+    // Storing all users in the leaderboard
+    // Passed down from the course component
+    @Input() leaderBoard: {
         name: string,
         token: number,
-    }];
+    } [];
 
-    // Table data
-    displayedColumns: string[] = ['name', 'token'];
+    displayedColumns = ['rank', 'name', 'token'];
     leaderBoardData: MatTableDataSource<{
         name: string,
         token: number,
     }>;
     @ViewChild(MatSort) matSort: MatSort;
 
-    constructor() {
-        this.leaderBoard = [{
-            name: 'Anonymous User',
-            token: null,
-        }];
-    }
-
     ngOnInit(): void {
-        this.leaderBoardData = new MatTableDataSource();
+
+        this.leaderBoardSort(this.leaderBoard);
+        this.podiumSort();
     }
 
     ngAfterViewInit(): void {
+        // initialize the datasource for the matTable from leaderBoard
         this.leaderBoardData = new MatTableDataSource(this.leaderBoard);
-        this.leaderBoardData.sort = this.matSort;
     }
-}
+
+    /**
+     * Sort the input leaderboard in descending order based on number of tokens
+     * @param leaderBoard - array of user, token pairs to be sorted
+     */
+    leaderBoardSort(leaderBoard: { name: string, token: number } []): void {
+        this.leaderBoard = leaderBoard.sort((a, b) => {
+            if (a.token < b.token) {
+                return 1;
+            }
+            if (a.token > b.token) {
+                return -1;
+            }
+            return 0;
+        });
+    }
+
+    /**
+     * Gets the top three teams on this leaderboard
+     */
+    podiumSort(): void {
```

Comment:
the function name is not correct. It doesn't sort anything. I recommend something like `calculateTopThree` or any better name you might think of.